tests/plugins/org.polarsys.capella.test.semantic.ui.ju/src/org/polarsys/capella/test/semantic/ui/ju/testcases/SemanticBrowserReferencingElementNavigationTest.java

```diff
@@ -147,7 +151,76 @@ public void test() throws Exception {
 
       IEditorPart editor = EclipseUIUtil.getActiveEditor();
       assertTrue("We should have a DDiagramEditor", editor instanceof DDiagramEditor);
+      
+      //Test that double-clicking an element (Capability here) opends a property wizard
```

Comment:
Here we're trying to do an integration test. Would it be better to do some Unit tests instead? 
For example, an idea would be:
- Separate the logic that handles the double-click into handler classes (e.g. a generic interface ISemanticBrowserDoubleClickingHandler and 2 classes inherting that interface ElementDoubleClickingHandler and DiagramDoubleClickingHandler)
- Add a new doubleClickingHandler field to the SemanticBroswerView class
- Change the handleDoubleClick method to associate the doubleClickingHandler field to the correct handler depending on clicked element and call the handler eventually.
- Now in the test we can just test that the handler is the correct instance depending on what's selected.
- We can go further to unit test the handlers to see if they really open the wizard and the diagram but I think it's not the scope of this test.

Reply:
I have tried to follow your approach but couldn't properly extract the different behavior into specific handlers
Since on a multiple selection we iterate over each element, the logic may differ .
Instead, to be able to do unit tests, I used boolean methods to know what can action is achieved when double-clicking a given element